### DISCUSSION TITLE
Stuck with Golang Hello World

### DISCUSSION THREAD

**Discussion by vikramchhibberTPM** [uber/cadence#5484] [Q&A]
Hi,

I am stuck with Golang very first hello world example. On my Mac desktop, I am successfully able to start server, installed cli, registered domain and verified the domain. But when I run the worker with workflow, I never see logs from the activity. I see following logs with exception.

2023-12-17T18:05:05.732-0800	WARN	internal/internal_worker.go:216	unable to verify if domain exist	{"Domain": "test-domain", "TaskList": "test-worker", "WorkerID": "61798@vikrams-MacBook-Pro.local@test-worker@273f7c44-d16b-4f70-a158-5fc4c6a62282", "WorkerType": "LocallyDispatchedActivityWorker", "domain": "test-domain", "error": "code:unavailable message:connection closed"}
go.uber.org/cadence/internal.verifyDomainExist.func1
	/usr/local/go/bin/pkg/mod/go.uber.org/cadence@v1.2.6/internal/internal_worker.go:216
go.uber.org/cadence/internal/common/backoff.Retry
	/usr/local/go/bin/pkg/mod/go.uber.org/cadence@v1.2.6/internal/common/backoff/retry.go:101
go.uber.org/cadence/internal.verifyDomainExist
	/usr/local/go/bin/pkg/mod/go.uber.org/cadence@v1.2.6/internal/internal_worker.go:226
go.uber.org/cadence/internal.(*activityWorker).Start
	/usr/local/go/bin/pkg/mod/go.uber.org/cadence@v1.2.6/internal/internal_worker.go:505
go.uber.org/cadence/internal.(*aggregatedWorker).Start
	/usr/local/go/bin/pkg/mod/go.uber.org/cadence@v1.2.6/internal/internal_worker.go:826
main.startWorker
	/Users/vikramchhibber/workspace/cadence/main.go:84
main.main
	/Users/vikramchhibber/workspace/cadence/main.go:28
runtime.main
	/usr/local/go/src/runtime/proc.go:267
2023-12-17T18:05:05.732-0800	INFO	internal/internal_worker.go:835	Started Activity Worker	{"Domain": "test-domain", "TaskList": "test-worker", "WorkerID": "61798@xyz-MacBook-Pro.local@test-worker@273f7c44-d16b-4f70-a158-5fc4c6a62282"}
2023-12-17T18:05:05.732-0800	INFO	cadence/main.go:89	Started Worker.	{"worker": "test-worker"}

docker run --network=host --rm ubercadence/cli:master --do test-domain workflow la

       WORKFLOW TYPE      |             WORKFLOW ID              |                RUN ID                |  TASK LIST  | IS CRON | START TIME | EXECUTION TIME | END TIME | CLOSE STATUS | HISTORY LENGTH | UPDATE TIME  
  main.helloWorldWorkflow | ccb2326a-96e1-40d8-9e5d-8b7c95c6a2ee | 483e9639-a148-45a4-8995-c2143aeaa7ac | test-worker | false   | 01:41:50   | 01:41:50       | 01:42:50 | TIMED_OUT    |              3 | 00:00:00     

docker run --network=host --rm ubercadence/cli:master --do test-domain domain describe
Name: test-domain
UUID: 10296bb1-0ceb-457d-ae3b-f00e76f90c27
Description: 
OwnerEmail: 
DomainData: map[]
Status: REGISTERED
RetentionInDays: 1
EmitMetrics: true
IsGlobal(XDC)Domain: true
ActiveClusterName: cluster0
Clusters: [cluster0]
HistoryArchivalStatus: DISABLED
VisibilityArchivalStatus: DISABLED

Please let me know if anyone has inputs.

Regards

**Comment by MatKurianski** [comment]
Same here :/

**Comment by 3vilhamster** [comment]
Hey.
Looking into the error, `code:unavailable message:connection closed` means that your service could not reach Cadence backend, so you need to expose ports, or start your service in the same network that Cadence is running in.

You can debug this by starting your service and checking if your service is registered on the backend side with 
`docker run --network=host --rm ubercadence/cli:master --do test-domain tl desc --tl test-worker`.

Do you have a sample service somewhere? May be I can spot check your configuration. 
Or maybe try running sample workflow and see what is missing https://github.com/uber-common/cadence-samples

**Comment by MatKurianski** [comment]
Thanks for the response @3vilhamster,

I am using the same [docker-compose.yml](https://github.com/uber/cadence/blob/master/docker/docker-compose.yml) referenced in the [documentation](https://cadenceworkflow.io/docs/get-started/installation/).

Then, I ran `docker compose up` and everything started as expected. I can even access Cadence UI and Grafana.

About the service, I copied and pasted the Golang hello world from the [documentation](https://cadenceworkflow.io/docs/get-started/golang-hello-world/#prerequisite). I put the code in [this gist](https://gist.github.com/MatKurianski/649f8ec001cd3a961e7276ae7c1495f9) just in case, but it should be the exactly same code of the documentation.

After some debugging, I believe the documentation has a typo: compared to cadence-samples repository, the documentation code listens to port 7933 instead of port 7833. 

I am new to Cadence so I am not entirely sure, but changing it to 7833 works as expected.
